src/lib/controllers/aligned-controller.js

```diff
@@ -11,23 +15,54 @@ export default class AlignedController {
       console.error(L10nSingleton.getMsgS('ALIGNED_CONTROLLER_NOT_READY_FOR_TOKENIZATION'))
       return false
     }
+
     this.alignment = alignment
     const tokenizer = 'simpleWordTokenization'
-    return this.alignment.createAlignedTexts(tokenizer)
+    const result = this.alignment.createAlignedTexts(tokenizer)
+
+    const res2 = this.alignment.equalSegmentsAmount
+
+    if (!res2) {
+      console.error(L10nSingleton.getMsgS('ALIGNED_CONTROLLER_NOT_EQUAL_SEGMENTS'))
+      this.alignment.clearAlignedTexts()
+      this.store.commit('incrementAlignmentUpdated')
+      return false

```

Comment:
@kirlat , I didn't finally understand what do you mean here saying Js constructor?
This is a method:
```
  createAlignedTexts (alignment) {
    if (!alignment || !alignment.readyForTokenize) {
      console.error(L10nSingleton.getMsgS('ALIGNED_CONTROLLER_NOT_READY_FOR_TOKENIZATION'))
      return false
    }

    this.alignment = alignment
    const tokenizer = 'simpleWordTokenization'
    const result = this.alignment.createAlignedTexts(tokenizer)

    const res2 = this.alignment.equalSegmentsAmount

    if (!res2) {
      console.error(L10nSingleton.getMsgS('ALIGNED_CONTROLLER_NOT_EQUAL_SEGMENTS'))
      this.alignment.clearAlignedTexts()
      this.store.commit('incrementAlignmentUpdated')
      return false
    }
    this.store.commit('incrementAlignmentUpdated')
    return result
  }
```
And it will always return Boolean value, true - if texts were aligned successfully, false - aligned texts were not created

Reply:
Oh, sorry, I was confused by the way the code changes were represented by the GitHub and I thought this code was part of a constructor. Now I realize that I was wrong. So please ignore my comment, there is absolutely no issue with the return value then.